src/mesh/mesh_base.C

```diff
@@ -335,20 +366,11 @@ void MeshBase::prepare_for_use (const bool skip_renumber_nodes_and_elements, con
   // Renumber the nodes and elements so that they in contiguous
   // blocks.  By default, _skip_renumber_nodes_and_elements is false.
   //
-  // We may currently change that by passing
-  // skip_renumber_nodes_and_elements==true to this function, but we
-  // should use the allow_renumbering() accessor instead.
-  //
   // Instances where you if prepare_for_use() should not renumber the nodes
   // and elements include reading in e.g. an xda/r or gmv file. In
   // this case, the ordering of the nodes may depend on an accompanying
   // solution, and the node ordering cannot be changed.
 
-  if (skip_renumber_nodes_and_elements)
-    {
-      libmesh_deprecated();
-      this->allow_renumbering(false);
```

Comment:
@roystgnr you're talking about how the new code makes permanent settings while the old code did not. I agree on the `find_neighbors` front. But this is a permanent setting, isn't it? And what's worse to me is that even if a user *explicitly* passes `skip_renumber_nodes_and_elements=false`, if someone in the past called it with `skip_renumber_nodes_and_elements`, then the new request is not going to be honored

Reply:
Moreover, we would give a deprecation warning to the first caller who passed `skip_renumber_nodes_and_elements=true` and  then honor his request. But for the guy who actually passes `skip_renumber_nodes_and_elements=false`, we give no warning at all before we potentially proceed to not do what he asks. (And I understand why we give no warning in the second case since it's a default argument...we have no way to know whether `skip_renumber_nodes_and_elements=false` has been explicitly requested or we're getting the default)

Reply:
Oh, wow, you're right.  What an annoyingly inconsistent old behavior.  I'm not thrilled with the new APIs either (since temporary changes to the find_neighbors setting now require a variable and a call to save the old setting, then the prepare_for_use, then a call to restore the old setting), but I can't help but admit that the old API was worse.